File: SecondYear/aayushker_AayushkerSingh_2226CSE(AI)1010_2/Week_4/First_Missing_Positive.py
from os import *
from sys import *
from collections import *
from math import *

def firstMissing(arr, n):
    # Write your function here.

    for i in range(n):
        while 1 <= arr[i] <= n and arr[arr[i] - 1] != arr[i]:
            arr[arr[i] - 1], arr[i] = arr[i], arr[arr[i] - 1]

    for i in range(n):
        if arr[i] != i + 1:
            return i + 1

    return n + 1
    pass

# Main Code
t=int(input())

for j in range(t):
    n=int(input())
    arr=[int(i) for i in input().split()]
    ans = firstMissing(arr,n)

    print(ans)

File: SecondYear/adityshrma08_AdityaSharma_2226csit1014_2/Week_4/First Missing Positive.cpp
#include <bits/stdc++.h> 
int firstMissing(int arr[], int n)
{
    // Write your code here.
    vector<int>v(n);

    for(int i=0;i<n;i++){
        v[i] = arr[i];
    }

    sort(v.begin(), v.end());
    int i=0,j=0;
    for(i=0;i<n;i++){
        if(v[i] > 0){
            j++;
            if(v[i]!=j)
            return j;
        }
    }
    return j+1;
}